Update Releasing section of README

coopergillan opened the pull request:
### What

* Update the documentation for creating a release
* Remove `make dist` as a dependency of `make release`

### Why

* With the Github Actions workflow added in dcc1d69, release binaries are now generated automatically when a tag is pushed, so the documentation should reflect this.
* The CHANGELOG generation taken care of by `make release` should be the last step before pushing the tag, which will then generate and publish the binaries, so nothing else need run before this.

@raymondberg

coopergillan committed:
Update Releasing section of README

What

Update the documentation for creating a release.

Why

With the Github Actions workflow added in dcc1d69, release binaries are
now generated automatically when a tag is pushed.

coopergillan commented:
@raymondberg - I am actually wondering if we even need `scripts/dist.sh` anymore. I suppose it wouldn't hurt to keep it for manually releasing, but do you think we could also get rid of that?

raymondberg reviewed (approved):
Agreed, and yeah we can remove dist

coopergillan commented:
@raymondberg - sure, though I'm realizing that, if for any reason you need to generate the binaries locally, you will still the script. Gonna leave it for now but we can revisit when and if need be.

coopergillan commented:
Looking again and it's actually `script/release.sh` that we shouldn't need anymore.

coopergillan committed:
Remove dist as GNUMakefile dependency for releasing

What

Remove the `make dist` run upon running `make release`.

Why

The actual binary building and release process is now taken care of by a
Github Actions per dcc1d69c9c36e70ef7b3ae7d970b8f359b23dfdb. While it
may still be necessary or desired to manually build the binaries, `make
release` now need only generate the CHANGELOG in preparation for pushing
the tag.

coopergillan commented:
Okay, ended up just removing a `GNUMakefile` dependency so that binaries can still be manually generated and so that the CHANGELOG generation is also preserved as the last step before publishing that release.
